include/ipc/ipc_service.h

```diff
@@ -70,20 +142,25 @@ struct ipc_ept_cfg {
 	void *priv;
 };
 
-/** @brief Register IPC endpoint.
+/** @brief Register IPC endpoint onto an instance.
  *
- *  Registers IPC endpoint to enable communication with a remote device.
+ *  Registers IPC endpoint onto an instance to enable communication with a
+ *  remote device.
  *
  *  The same function registers endpoints for both master and slave devices.
  *
  *  @param ept Endpoint object.
  *  @param cfg Endpoint configuration.
+ *  @param target_instance Name of the instance to register the endpoint onto.
+ *                         Can be NULL when the backend implements an internal
+ *                         logic for endpoint-instance allocation.
  *
  *  @retval -EIO when no backend is registered.
  *  @retval -EINVAL when pointer to an endpoint or endpoint configuration is invalid.
  *  @retval Other errno codes depending on the implementation of the backend.
  */
-int ipc_service_register_endpoint(struct ipc_ept **ept, const struct ipc_ept_cfg *cfg);
+int ipc_service_register_endpoint(struct ipc_ept **ept, const struct ipc_ept_cfg *cfg,
+				  const char *target_instance);
```

Comment:
Nitpicking: maybe it would be better to have a shorter parameter name for the indentation `instance` seems explicit enough.

Reply:
ACK